05-JWT-Basics/starter/controllers/main.js

```diff
@@ -0,0 +1,62 @@
+//check username, password in post(login) request
+//if exist create new JWT
+//send back to from-end
+
+//setup authentication so only the request with JWT can access the dashboard
+
+const jwt = require("jsonwebtoken");
+const { BadRequestError } = require("../errors");
+const login = async (req, res) => {
+  const { username, password } = req.body;
+  //mongoose validation
+  //Joi
+  //check in the controller
+
+  if (!username || !password) {
+    throw new BadRequestError("Please provide email and password");
```

Comment:
Minor note: It requests a username and password, so this error message is confusing. But this isn't a big deal, it works

Reply:
Thank you for your feedback and for reviewing the assignment.